packages/synapse-react-client/src/components/EntityViewScopeEditor/EntityViewScopeEditorModal.test.tsx

```diff
@@ -0,0 +1,235 @@
+      ...mockTableEntity,
+      scopeIds: ['syn123'],
+      viewTypeMask: ENTITY_VIEW_TYPE_MASK_FILE | ENTITY_VIEW_TYPE_MASK_FOLDER,
+      concreteType: ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+    } as Entity)
+  })
+
+  it('displays the correct scope editor and mask editor after initial render', async () => {
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    expect(
+      within(dialog).getByRole('heading', { name: 'Edit Scope' }),
+    ).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Cancel' })).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Save' })).toBeVisible()
+
+    const filesButton = within(dialog).getByRole('checkbox', { name: 'Files' })
+    const foldersButton = within(dialog).getByRole('checkbox', {
+      name: 'Folders',
+    })
+    const tablesButton = within(dialog).getByRole('checkbox', {
+      name: 'Tables',
+    })
+    const datasetsButton = within(dialog).getByRole('checkbox', {
+      name: 'Datasets',
+    })
+
+    const checkboxes = screen.getAllByRole('checkbox')
+    expect(checkboxes).toHaveLength(4)
+
+    expect(filesButton).toBeVisible()
+    expect(foldersButton).toBeVisible()
+    expect(tablesButton).toBeVisible()
+    expect(datasetsButton).toBeVisible()
+
+    await waitFor(() => {
+      expect(filesButton).toBeChecked()
+      expect(foldersButton).toBeChecked()
+      expect(tablesButton).not.toBeChecked()
+      expect(datasetsButton).not.toBeChecked()
+    })
+  })
+
+  it('display error for invalid entity', async () => {
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    mockUpdateEntity.mockRejectedValue(
+      new SynapseClientError(
+        404,
+        'Error with scope',
+        expect.getState().currentTestName!,
+      ),
+    )
+
+    const saveButton = within(dialog).getByRole('button', { name: 'Save' })
+    await waitFor(() => {
+      expect(
+        within(dialog).getByRole('button', { name: 'Save' }),
+      ).not.toBeDisabled()
+    })
+    await userEvent.click(saveButton)
+    expect(mockOnUpdate).not.toHaveBeenCalled()
+  })
+
+  it('verify no mask options are shown when projectView is passed', async () => {
+    jest.clearAllMocks()
+
+    jest.spyOn(SynapseClient, 'getEntity').mockResolvedValue({
+      ...mockTableEntity,
+      scopeIds: ['syn123'],
+      viewTypeMask: ENTITY_VIEW_TYPE_MASK_PROJECT,
+      concreteType: ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+    } as Entity)
+
+    renderComponent({
+      entityId: mockProjectViewEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
```

Comment:
Not that it matters, since you mocked `getEntity`, to always return it, but I would expect to see `mockProjectViewEntity` instead of `mockTableEntity`--it is kind of weird to pass one ID as props but have your mock service return a completely different entity. It might confuse someone working on this test in the future.